src/lib/bootloader/systemdboot.rb

```diff
@@ -0,0 +1,304 @@
+                     "Error output: %{stderr}"
+                   ), command: e.commands.inspect, stderr: e.stderr)
+          )
+            break
+          end
+        end
+      end
+      File.delete(cmdline_file) if Yast::Stage.initial
+    end
+
+    def read_menue_timeout
+      filename = File.join(Yast::Installation.destdir, LOADERCONF)
+      return unless File.file?(filename)
+
+      f = File.open(filename, "r")
+      f.each_line do |line|
+        line.lstrip!
+        next if line.start_with?("#")
+
+        if line.start_with?("timeout")
+          self.menue_timeout = line.split[1].to_i
+          break
+        end
+      end
+    end
+
+    def write_menue_timeout
+      filename = File.join(Yast::Installation.destdir, LOADERCONF)
+      return unless File.file?(filename)
+
+      output = []
+      entry_set = false
+      f = File.open(filename, "r")
+      f.each_line do |line|
+        line.lstrip!
+        if line.start_with?("timeout") || line.start_with?("#timeout")
+          output << "timeout #{menue_timeout}\n"
+          entry_set = true
+          next
+        end
+        output << line
+      end
+      output << "timeout #{menue_timeout}" unless entry_set
+      File.open(filename, "w+") do |fw|
+        fw.puts(output)
+      end
+    end
+
+    def bootloader_is_installed
+      Yast::Execute.on_target(BOOTCTL, "is-installed", allowed_exitstatus: 1) == 0
+    end
+
+    def remove_secure_boot_settings
+      del_files = ["/boot/efi/EFI/systemd/grub.efi",
+                   "/boot/efi/EFI/systemd/systemd-bootx64.efi",
+                   "/boot/efi/EFI/systemd/MokManager.efi"]
+      del_files.each do |f|
+        filename = File.join(Yast::Installation.destdir, f)
+        File.delete(filename) if File.exist?(filename)
+      end
+    end
+
+    def secure_boot_available
+      ret = false
+      begin
+        ret = Yast::Execute.on_target(MOKUTIL, "--sb-state", allowed_exitstatus: 1) == 0
+      rescue Cheetah::ExecutionFailed => e
+        log.info("Command `#{e.commands.inspect}`.\n" \
+                 "Error output: #{e.stderr}")
+      end
+      ret
+    end
+
+    def delete_bootloader
+      return unless bootloader_is_installed
+
+      log.info("Removing already installed systemd bootmanager.")
+      begin
+        Yast::Execute.on_target!(BOOTCTL, "remove")
+      rescue Cheetah::ExecutionFailed => e
+        Yast::Report.Error(
+        format(_(
+               "Cannot remove systemd bootloader:\n" \
+               "Command `%{command}`.\n" \
+               "Error output: %{stderr}"
+             ), command: e.commands.inspect, stderr: e.stderr)
+      )
+        return
+      end
+      remove_secure_boot_settings
+    end
+
+    def set_secure_boot
+      log.info("Enabling secure boot options")
+      src = File.join(Yast::Installation.destdir, "/boot/efi/EFI/systemd/systemd-bootx64.efi")
+      dest = File.join(Yast::Installation.destdir, "/boot/efi/EFI/systemd/grub.efi")
```

Comment:
hmm, this will need some comments **why** mv is needed and why we cannot use default systemd file name.

Reply:
Unfortunately, shim currently only checks hardcoded for grub.efi.
See also https://en.opensuse.org/Systemd-boot#Secure_Boot

Reply:
ok, so @schubi2 please document rationale for this step in comment in code, otherwise in few years we will forget it and maybe when shim start supporting it, we will be able to remove it.
Thanks

Reply:
I have described it.